kv/raftstore/peer_storage.go

```diff
@@ -506,64 +447,60 @@ func (ps *PeerStorage) ApplySnapshot(ctx *InvokeContext, snap *eraftpb.Snapshot,
 /// Do not modify ready in this function, this is a requirement to advance the ready object properly later.
 func (ps *PeerStorage) SaveReadyState(ready *raft.Ready) (*ApplySnapResult, error) {
 	kvWB, raftWB := new(engine_util.WriteBatch), new(engine_util.WriteBatch)
-	ctx := NewInvokeContext(ps)
+	prevRaftState := ps.raftState
+
 	var snapshotIdx uint64 = 0
+	var snapRegion *metapb.Region = nil
+	var applyRes *ApplySnapResult = nil
 	if !raft.IsEmptySnap(&ready.Snapshot) {
-		if err := ps.ApplySnapshot(ctx, &ready.Snapshot, kvWB, raftWB); err != nil {
+		applySate, region, err := ps.ApplySnapshot(&ready.Snapshot, kvWB, raftWB)
+		if err != nil {
 			return nil, err
 		}
-		snapshotIdx = ctx.RaftState.LastIndex
-		ctx.saveApplyStateTo(kvWB)
+		applyRes = &ApplySnapResult{
+			PrevRegion: ps.region,
+			Region:     region,
+		}
+		snapshotIdx = ps.raftState.LastIndex
+		snapRegion = region
+		ps.saveApplyStateTo(kvWB, applySate)
 	}
 
 	if len(ready.Entries) != 0 {
-		if err := ps.Append(ctx, ready.Entries, raftWB); err != nil {
+		if err := ps.Append(ready.Entries, raftWB); err != nil {
 			return nil, err
 		}
 	}
 
-	// Last index is 0 means the peer is created from raft message
-	// and has not applied snapshot yet, so skip persistent hard state.
-	if ctx.RaftState.LastIndex > 0 {
```

Comment:
why remove this

Reply:
Seems it just a optimization only works one time, but the comment is quite confusing as the `RaftInitLogIndex = 5`.